### PR TITLE
Add support for UP2 board

### PR DESCRIPTION
**Pull request by thien94**
The instructions in this PR will allow user to setup an APSync image for the Up2 board like other boards, with additional support for the Realsense T265 and D435 cameras.

The new features follow these blog posts: [1](https://discuss.ardupilot.org/t/gsoc-2020-enhancements-of-non-gps-navigation-and-object-avoidance-with-realsense-cameras/55984), [2](https://discuss.ardupilot.org/t/gsoc-2020-integration-of-ardupilot-and-realsense-d4xx-depth-camera-for-simple-obstacle-avoidance/58474/1), [3](https://discuss.ardupilot.org/t/gsoc-2020-enhancements-of-gps-less-navigation-and-obstacle-avoidance-with-tracking-and-depth-camera/60285).

The PR has also been tested by @patrickpoirier51.

### CODE REVIEW

File: Common/Ubuntu/librealsense/d4xx.service [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=Realsense D4xx Service
+After==multi-user.target
+StartLimitIntervalSec=0
+Conflicts=
+
+[Service]
+User=apsync
+EnvironmentFile=
```
Comment (rmackay9) [inline Common/Ubuntu/librealsense/d4xx.service:9]: I really don't know what I'm talking about but I guess these lines are important?  I.e. "EnvironmentFile=" with no path is required.

File: Up_Squared/Ubuntu/2_Clone_Repo_Disable_console_sethost.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+apt autoremove -y # avoid repeated no-longer-required annoyance
+
+apt-get remove -y unattended-upgrades
+
+tput setaf 3
+echo "Setting up rc.local"
+tput sgr0
+./ensure_rc_local.sh
+
+# tput setaf 3
```
Comment (rmackay9) [inline Up_Squared/Ubuntu/2_Clone_Repo_Disable_console_sethost.sh:40]: Are these commented out lines required?  never used?  sometimes used?

File: Up_Squared/Ubuntu/4_setup_apsync_components.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+sudo -u $NORMAL_USER -H bash <<EOF
+   set -e
+   set -x
+
+   pushd /home/$NORMAL_USER/GitHub/pymavlink
+   git config --global user.email "devel@ardupilot.org"
+   git config --global user.name "ArduPilotCompanion"
+
+   git stash
+   git revert e1532c3fc306d83d03adf82fb559f1bb50860c03
```
Comment (rmackay9) [inline Up_Squared/Ubuntu/4_setup_apsync_components.sh:36]: I don't think this is actually required anymore although I'm not 100% sure... but I see that I've left them in the RPI equivalent so let's leave them here too, thanks.

File: Up_Squared/Ubuntu/install_wifi_access_point.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+# NOTE: if something goes wrong with this setup, and you are unable to connect to the TX1, connect a keyboard and mouse and type "sudo dhclient eth0"
+apt-get -y install dnsmasq haveged
+apt-get -y install wpasupplicant
+
+# stop dnsmasq from running outside of where we want it to:
+sudo systemctl disable dnsmasq
+
+# Create Access Point
+SSID="ardupilot"
```
Comment (rmackay9) [inline Up_Squared/Ubuntu/install_wifi_access_point.sh]: On the RPI we set the SSID to "ArduPilot" with a capital A and P..

**Comment by rmackay9** [comment]
Looking pretty good to me, thanks very much for this!

File: Common/Ubuntu/librealsense/d4xx.service (hunk 1 above)
Comment (thien94) [reply Common/Ubuntu/librealsense/d4xx.service:9]: For this I am using the same format as the .service file in the RPi2 folder. I don't think it is required when there is no path, but we can just leave it there for consistency and to reduce ambiguity (either default path or a defined path)?

File: Up_Squared/Ubuntu/2_Clone_Repo_Disable_console_sethost.sh (hunk 2 above)
Comment (thien94) [reply Up_Squared/Ubuntu/2_Clone_Repo_Disable_console_sethost.sh:40]: It is not required, I think I can remove them if need to.

**Comment by Hwurzburg** [comment]
@rmackay9 approve the changes and then we can merge...thanks

File: Common/Ubuntu/librealsense/d4xx.service (hunk 1 above)
Comment (rmackay9) [reply Common/Ubuntu/librealsense/d4xx.service:9]: ok!

File: Up_Squared/Ubuntu/install_wifi_access_point.sh (hunk 4 above)
Comment (rmackay9) [reply Up_Squared/Ubuntu/install_wifi_access_point.sh]: txs for fixing this

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**